src/store/modules/accounts/hdWallet.js

```diff
@@ -30,16 +31,19 @@ export default {
       return aeSdk.api.getAccountByPubkey(address).then(() => true, () => false);
     },
     async discover({ rootGetters: { wallet }, dispatch }) {
-      const numberOfAccounts = await ProtocolAdapterFactory
-        .getAdapter(PROTOCOL_AETERNITY) // Discover only aeternity accounts for now
-        .discoverAccounts(wallet);
-
-      for (let i = 0; i < numberOfAccounts; i += 1) {
-        dispatch('create', {
-          isRestored: true,
-          protocol: PROTOCOL_AETERNITY,
-        });
-      }
+      const accountsToRecover = await Promise.all(
+        PROTOCOLS.map(async (protocol) => (
+          await ProtocolAdapterFactory.getAdapter(protocol)
+        ).discoverAccounts(wallet)),
+      );
+      accountsToRecover.forEach(({ protocol, lastNotEmptyIdx }) => {
```

Comment:
As I think the `accountsToRecover` don't need to hold the protocol name - only the number, so here instead of looping the `accountsToRecover` you can loop again the `PROTOCOLS` constant. I did this as a part of the PR that covers the migration of accounts from Vuex to composable: https://github.com/aeternity/superhero-wallet/pull/2334/files#diff-10c662816c7957770974da4b39e9e9a3ca2135b2fa50d7bba66c45f62a9b369cR196